### DISCUSSION TITLE
Help with rule IDE1006 "Naming rule violation: Missing prefix: '____RULE_VIOLATION____'"

### DISCUSSION THREAD

**Discussion by GerryWilko** [RehanSaeed/EditorConfig#37] [Q&A]
Hello,

I have recently added your .editorconfig file to our project. Thank you for taking the time to produce this, its fantastic and encourages best practices.

I am having a rather strange issue however. Rule IDE1006 is attempting to prefix and suffix all my uses of protected variables throughout my codebase. Strangely a google of this exact error message reveals nothing which would suggest no one else has experienced this issue.

The lines in question do conform to the .editorconfig rules. For example:

    protected PolicyService policyService;

But they produce the Error IDE1006 "Naming rule violation: Missing prefix: '____RULE_VIOLATION____'"

Applying the autofix for it produces the following line:

    protected PolicyService ____RULE_VIOLATION____PolicyService____RULE_VIOLATION____;

This is obviously wrong but I cant see what I am doing wrong here. Does anybody have any ideas?

Many thanks,
Gerard

**Comment by GerryWilko** [comment]
I read through the rules again and I've sussed it. This rather unhelpful message is telling me that I cannot have non-private instance fields. Make sense, you should always use C# properties for this.

Hope this helps someone else out. Took me a while to decipher.

    protected PolicyService PolicyService { get; }

**Comment by RehanSaeed** [comment]
Glad you sorted it. Unfortunately `.editorconfig` doesn't give us a way to provide a clean error message for custom rules that we've setup like this one.
